File: src/main/java/pd/discordipc/utils/TimeUtils.java
package pd.discordipc.utils;

public class TimeUtils {
    public static long parseToSeconds(String time) {
        if (time == null || !time.contains(":")) return 0;
        String[] parts = time.split(":");
        try {
            if (parts.length == 2) {
                return Long.parseLong(parts[0]) * 60 + Long.parseLong(parts[1]);
            } else if (parts.length == 3) {
                return Long.parseLong(parts[0]) * 3600 + Long.parseLong(parts[1]) * 60 + Long.parseLong(parts[2]);
            }
        } catch (NumberFormatException e) {
            return 0;
        }
        return 0;
    }
}

File: src/main/kotlin/com/pdlayer/discordipc/utils/TimeUtils.kt
package com.pdlayer.discordipc.utils

object TimeUtils {
    private val HOURS_MINUTES_SECONDS = Regex("""(\d{1,4}):([0-5]\d):([0-5]\d)""")
    private val MINUTES_SECONDS = Regex("""(\d{1,4}):([0-5]\d)""")

    fun parseToSeconds(time: String?): Long {
        if (time == null) {
            return 0
        }

        val value = time.trim()

        HOURS_MINUTES_SECONDS.matchEntire(value)?.let { match ->
            val (hours, minutes, seconds) = match.destructured
            return hours.toLong() * 3600 + minutes.toLong() * 60 + seconds.toLong()
        }

        MINUTES_SECONDS.matchEntire(value)?.let { match ->
            val (minutes, seconds) = match.destructured
            return minutes.toLong() * 60 + seconds.toLong()
        }

        throw IllegalArgumentException(
            "Time must be mm:ss or hh:mm:ss with minutes and seconds in 0..59, got \"$time\""
        )
    }
}
